### PR TITLE
chore: Prevent test output from being polluted with Icon exceptions

### PR DESCRIPTION
**Pull request by schne324**
## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Tested for accessibility
- [x] Code is reviewed for security

### CODE REVIEW

File: packages/react/src/components/Icon/index.tsx [hunk 1]
```diff
@@ -16,6 +16,9 @@ function Icon({ label, className, type, ...other }: IconProps) {
   const [IconSVG, setIcon] = useState<React.ComponentType<any> | null>(null);
 
   useEffect(() => {
+    if (process.env.JEST_WORKER_ID !== undefined) {
```
Comment (stephenmathieson) [inline packages/react/src/components/Icon/index.tsx:19]: What if we checked `NODE_ENV` instead?

```suggestion
    if (process.env.NODE_ENV === "test") {
```

File: tsconfig.json [hunk 2]
```diff
@@ -9,7 +9,8 @@
     "strictNullChecks": true,
     "strictFunctionTypes": true,
     "typeRoots": ["node_modules/@types"],
-    "jsx": "react"
+    "jsx": "react",
+    "types": ["node"]
```
Comment (stephenmathieson) [inline tsconfig.json:13]: Don't need this. Should remove it.

```suggestion
```

Comment (schne324) [reply tsconfig.json:13]: I'm pretty sure without it the [build script broke](https://app.circleci.com/pipelines/github/dequelabs/cauldron/180/workflows/7fcfcafd-4583-48af-8d3c-8b52021253cb/jobs/736) but I'll remove it and give it a try 🤷‍♂️ 

```
[!] (plugin typescript) Error: @rollup/plugin-typescript TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
src/components/Icon/index.tsx (21:9)
```

Comment (schne324) [reply tsconfig.json]: Yep. Removing `"types": ["node"]` broke the build: https://app.circleci.com/pipelines/github/dequelabs/cauldron/183/workflows/933ecaeb-99a4-47cc-941a-730939dcf718/jobs/747

Comment (stephenmathieson) [reply tsconfig.json]: Strange.. might have something to do with setting `typeRoots` (which is unnecessary most of the time), or using Babel to compile TypeScript. It's probably worth looking into this in the future. Not going to block this PR for that tho.

**Review by stephenmathieson — APPROVED** [review]

**Review by stephenmathieson — APPROVED** [review]

**Pull request merged**